fish-rust/src/env_dispatch.rs

```diff
@@ -0,0 +1,809 @@
+            "True-color support: enabled per termcap/terminfo for",
+            term,
+            "with",
+            max_colors.unwrap(),
+            "colors"
+        );
+    } else if let Some(ct) = vars.get(L!("COLORTERM")).map(|v| v.as_string()) {
+        // If someone sets $COLORTERM, that's the sort of color they want.
+        if ct == L!("truecolor") || ct == L!("24bit") {
+            supports_24bit = true;
+        }
+        FLOGF!(
+            term_support,
+            "True-color support",
+            if supports_24bit {
+                "enabled"
+            } else {
+                "disabled"
+            },
+            "per $COLORTERM",
+            ct
+        );
+    } else if vars.get(L!("KONSOLE_VERSION")).is_some()
+        || vars.get(L!("KONSOLE_PROFILE_NAME")).is_some()
+    {
+        // All Konsole versions that use $KONSOLE_VERSION are new enough to support this, so no
+        // check is needed.
+        supports_24bit = true;
+        FLOGF!(term_support, "True-color support: enabled for Konsole");
+    } else if let Some(it) = vars.get(L!("ITERM_SESSION_ID")).map(|v| v.as_string()) {
+        // Supporting versions of iTerm include a colon here.
+        // We assume that if this is iTerm it can't also be st, so having this check inside is okay.
+        if !it.contains(':') {
+            supports_24bit = true;
+            FLOGF!(term_support, "True-color support: enabled for iTerm");
+        }
+    } else if term.starts_with("st-") {
+        supports_24bit = true;
+        FLOGF!(term_support, "True-color support: enabling for st");
+    } else if let Some(vte) = vars.get(L!("VTE_VERSION")).map(|v| v.as_string()) {
+        if fish_wcstoi(vte.chars()).unwrap_or(0) > 3600 {
+            supports_24bit = true;
+            FLOGF!(
+                term_support,
+                "True-color support: enabled for VTE version",
+                vte
+            );
+        }
+    }
+
+    let support = if supports_256color {
+        ColorSupport::TERM_256COLOR
+    } else {
+        ColorSupport::NONE
+    } | if supports_24bit {
+        ColorSupport::TERM_24BIT
+    } else {
+        ColorSupport::NONE
+    };
+    unsafe {
+        crate::output::output_set_color_support(support.bits() as i32);
+    }
+}
+
+/// Try to initialize the terminfo/curses subsystem using our fallback terminal name. Do not set
+/// `$TERM` to our fallback. We're only doing this in the hope of getting a functional shell.
+/// If we launch an external command that uses `$TERM`, it should get the same value we were given,
+/// if any.
+fn initialize_curses_using_fallbacks(vars: &dyn Environment) {
+    // xterm-256color is the most used terminal type by a massive margin, especially counting
+    // terminals that are mostly compatible.
+    const FALLBACKS: [&str; 4] = ["xterm-256color", "xterm", "ansi", "dumb"];
+
+    let termstr = vars
+        .get_unless_empty(L!("TERM"))
+        .map(|v| v.as_string().to_string())
```

Comment:
Please do not use `to_string` here. It can just be removed.

Reply:
Yup. I'll change it the function to take a `CStr`.